### PR TITLE
Add configuration setting for TTL of HTTP connections to IRestfulClientFactory

### LINKED ISSUE #6184: Allow the TTL of HTTP connections made by the PoolingHttpClientConnectionManager created by ApacheRestfulClientFactory::getNativeHttpClient to be customizable
**Describe the issue**
My team currently uses the [ApacheRestfulClientFactory](https://github.com/hapifhir/hapi-fhir/blob/0397b9ddc8a8672d64fa726c26c13fbfc7b97a25/hapi-fhir-client/src/main/java/ca/uhn/fhir/rest/client/apache/ApacheRestfulClientFactory.java) provided by hapi-fhir-client.  I have a requirement to increase the time to live (TTL) of the HTTP connections that are made by our FHIR HTTP client.  It looks like the TTL of the connections made by the HTTP clients built by the ApacheRestfulClientFactory is controlled by this value, which is hard-coded to 5 seconds:
https://github.com/hapifhir/hapi-fhir/blob/0397b9ddc8a8672d64fa726c26c13fbfc7b97a25/hapi-fhir-client/src/main/java/ca/uhn/fhir/rest/client/apache/ApacheRestfulClientFactory.java#L106

The only way I can see to customize the TTL of the connections is to completely override the logic for building the HttpClient, which would be an undesirable solution.

Is it possible to allow the hard-coded 5 second value for the TTL of HTTP connections made by the HTTP clients built by ApacheRestfulClientFactory::getNativeHttpClient to be customized?  Without this ability, I would have to provide my own HttpClient implementation from scratch in order to modify the TTL of the HTTP connections it makes.

**Environment (please complete the following information):**
 - HAPI FHIR Version: 7.0.2
 - OS: Linux
 - Browser: N/A

### PR DESCRIPTION
**Pull request by acoteathn**
fixes #6184

### CODE REVIEW

**Comment by tadgh** [comment]
Hey there! This looks good! Thanks for the contribution. Only thing is, yes, please add a changelog in the `7_6_0` folder of the documentation project. It should follow the templates of the other changelogs, with `type: add` and the filename prefixed with the issue number (in this case 6184). Also, feel free to attribute yourself in the changelog, and add yourself to our <developers> section in our pom.xml. If you'd rather not do those parts, that's fine too. Or if you'd rather I do them, I can do that as well. Cheers!

File: pom.xml [hunk 1]
```diff
@@ -919,15 +919,21 @@
...
-    </developer>
+		</developer>
+		<developer>
+			<id>melihaydogd</id>
+			<name>Ahmet Melih Aydoğdu</name>
+		</developer>
+		<developer>
+			<id>alexrkopp</id>
+			<name>Alex Kopp</name>
+			<organization>athenahealth</organization>
```
Comment (acoteathn) [inline pom.xml:930]: Alex is a colleague of mine, so I added this line! I can remove it if people don't want me editing the entry on behalf of other people, though :)

**Comment by acoteathn** [comment]
Hi @tadgh, I've made the changes you mentioned

**Comment by acoteathn** [comment]
I also noticed there are some failing builds. I'm not sure if there's anything there which I need to take action on, though.

**Comment by tadgh** [comment]
Don't worry about that, I'm re-running them for you. The only ones that matter for merge purposes are the tests

**Pull request merged**